src/component/FileManager/ContextMenu/useActionDisplayOpt.ts

```diff
@@ -291,11 +293,23 @@ export const getActionOpt = (
     display.orCapability &&
     display.orCapability.enabled(NavigatorCapability.download_file);
 
+  // Reset thumbnail is available when at least one file is selected and
+  // current capability allows generating thumbnails
+  // Show only when at least one selected file has a supported extension,
+  // based on cached supported thumbnail extensions.
+  const cache = getCachedThumbExts();
+  const anySupported =
+    cache instanceof Set
+      ? targets.some((f) => f.type == FileType.file && cache.has((fileExtension(f.name) || "").toLowerCase()))
+      : false;
+  display.showResetThumb = display.hasFile && anySupported;
```

Comment:
Load thumb exts only after user clicked the context menu option.
Here you just need something like
```
 display.showResetThumb = display.hasFile  && has update permission && has thumb disabled key;
```